src/util/antlr/ANTLRErrorHandling.h

```diff
@@ -4,21 +4,54 @@
 
 #pragma once
 
+#include <parser/ParseException.h>
+
 #include <string>
 
-#include "../../parser/ParseException.h"
 #include "BaseErrorListener.h"
+#include "CharStream.h"
+#include "CommonTokenStream.h"
 #include "DefaultErrorStrategy.h"
+#include "ParserRuleContext.h"
 #include "RecognitionException.h"
+#include "Recognizer.h"
+#include "TokenSource.h"
+
+// TODO<qup42> whats up with this namespace?
+namespace {
+[[maybe_unused]] std::string colorError(const ExceptionMetadata& metadata) {
+  if (metadata.startIndex < metadata.stopIndex) {
+    auto first = metadata.query_.substr(0, metadata.startIndex);
+    auto middle = metadata.query_.substr(
+        metadata.startIndex, metadata.stopIndex - metadata.startIndex + 1);
+    auto end = metadata.query_.substr(metadata.stopIndex + 1);
 
-struct ThrowingErrorStrategy : public antlr4::DefaultErrorStrategy {
-  void reportError(antlr4::Parser*,
-                   const antlr4::RecognitionException& e) override {
-    throw antlr4::ParseCancellationException{
-        e.what() + std::string{" at token \""} +
-        e.getOffendingToken()->getText() + '"'};
+    return (first + "\x1b[1m\x1b[4m\x1b[31m" + middle + "\x1b[0m" + end);
+  } else {
+    return metadata.query_;
   }
-};
+}
+
+ExceptionMetadata generateMetadata(antlr4::Recognizer* recognizer,
+                                   antlr4::Token* offendingToken,
+                                   [[maybe_unused]] size_t line,
+                                   [[maybe_unused]] size_t charPositionInLine) {
+  // TODO<qup42> the source query differs from the real query. The SparqlLexer
+  //  lowercases some parts and sometimes adds whitespaces.
```

Comment:
Yes the ANTLR provided data is consistent. The highlight in the terminal is nice. I had hoped to also be able to do this in the UI. Imo this would have been a big improvement for error handling.

When the whole query is parsed with ANTLR this problem should fix itself.